packages/cardano-services/src/ChainHistory/DbSyncChainHistory/mappers.ts

```diff
@@ -172,6 +195,128 @@ export const mapCertificate = (
+        type: Number(certModel.has_script) ? Cardano.CredentialType.ScriptHash : Cardano.CredentialType.KeyHash
+      },
+      stakeCredential: {
+        hash: Cardano.RewardAccount.toHash(
+          Cardano.RewardAccount(certModel.address)
+        ) as unknown as Crypto.Hash28ByteBase16,
+        type: Cardano.CredentialType.KeyHash
+      }
+    };
+
+  if (isVoteRegistrationDelegationCertModel(certModel))
+    return {
+      __typename: Cardano.CertificateType.VoteRegistrationDelegation,
+      cert_index: certModel.cert_index,
+      dRep: {
+        hash: certModel.drep_hash as Hash28ByteBase16,
+        type: Number(certModel.has_script) ? Cardano.CredentialType.ScriptHash : Cardano.CredentialType.KeyHash
+      },
+      deposit: BigInt(certModel.deposit),
+      stakeCredential: {
+        hash: Cardano.RewardAccount.toHash(
+          Cardano.RewardAccount(certModel.address)
+        ) as unknown as Crypto.Hash28ByteBase16,
+        type: Cardano.CredentialType.KeyHash
+      }
+    };
+
+  if (isStakeVoteDelegationCertModel(certModel))
+    return {
+      __typename: Cardano.CertificateType.StakeVoteDelegation,
+      cert_index: certModel.cert_index,
+      dRep: {
+        hash: certModel.drep_hash as Hash28ByteBase16,
+        type: Number(certModel.has_script) ? Cardano.CredentialType.ScriptHash : Cardano.CredentialType.KeyHash
+      },
+      poolId: certModel.pool_id as unknown as Cardano.PoolId,
+      stakeCredential: {
+        hash: Cardano.RewardAccount.toHash(
+          Cardano.RewardAccount(certModel.address)
+        ) as unknown as Crypto.Hash28ByteBase16,
+        type: Cardano.CredentialType.KeyHash
+      }
+    };
+
+  if (isStakeRegistrationDelegationCertModel(certModel))
+    return {
+      __typename: Cardano.CertificateType.StakeRegistrationDelegation,
+      cert_index: certModel.cert_index,
+      deposit: BigInt(certModel.deposit),
+      poolId: certModel.pool_id as unknown as Cardano.PoolId,
+      stakeCredential: {
+        hash: Cardano.RewardAccount.toHash(
+          Cardano.RewardAccount(certModel.address)
+        ) as unknown as Crypto.Hash28ByteBase16,
+        type: Cardano.CredentialType.KeyHash
+      }
+    };
+
+  if (isStakeVoteRegistrationDelegationCertModel(certModel))
+    return {
+      __typename: Cardano.CertificateType.StakeVoteRegistrationDelegation,
+      cert_index: certModel.cert_index,
+      dRep: {
+        hash: certModel.drep_hash as Hash28ByteBase16,
+        type: Number(certModel.has_script) ? Cardano.CredentialType.ScriptHash : Cardano.CredentialType.KeyHash
+      },
+      deposit: BigInt(certModel.deposit),
+      poolId: certModel.pool_id as unknown as Cardano.PoolId,
+      stakeCredential: {
+        hash: Cardano.RewardAccount.toHash(
+          Cardano.RewardAccount(certModel.address)
+        ) as unknown as Crypto.Hash28ByteBase16,
+        type: Cardano.CredentialType.KeyHash
+      }
+    };
+
+  /**
+   * TODO: Can these be returned from db-sync?
```

Comment:
Can we link to the issue raised against `db-sync` so that we can track it?

Reply:
a6d11ce74f should solve this